### ISSUE TITLE
Inventory can be opened before game is loaded 

### ISSUE THREAD

**Issue by burner1024** [gemrb/gemrb#1992]
#### Bug description
Inventory screen can be opened during loading game. That results in un-paused inventory state. Closing the inventory then pauses the game.

#### Steps to reproduce
1. Load game.
2. During loading, tap I, maybe a few times.
3. Inventory opens before loading screen is closed, game continues to tick.

#### Expected behavior
No GUI interaction possible until loading is completed and loading screen is closed.

#### Screenshots

https://github.com/gemrb/gemrb/assets/1279852/7f40ad89-79ee-4bf4-afed-76a36a1372ae

GemRB version (check as many as you know apply)
- [x] master as of this issue
- [ ] 0.9.2
- [ ] 0.9.1
- [ ] 0.9.0

Video Driver (check as many as you know apply)
- [ ] SDL1.2
- [x] SDL2 built with `USE_OPENGL_BACKEND`
- [ ] SDL2 without `USE_OPENGL_BACKEND`

**Comment by bradallred** [comment]
Probably means this applies to other UIs too. We should find a general fix. We should have a way to know when we are still loading. At worst checking the load screen variable or for the windows itself is doable.

**Comment by lynxlynxlynx** [comment]
Yeah, it would be easy to check "Progress" in the subfunction of CreateTopWinLoader.

**Comment by burner1024** [comment]
Ideally game clock should start to tick at the exact same moment.

**Comment by bradallred** [comment]
Glancing at the code, the GUI is the last thing to come up, so adding something to check the state shouldn't be necessary. 

does this only happen when loading a game from within another game? I suspect the GUI being already loaded would be the path to reproducing this.

I think something close to this would be what we want:

```diff
diff --git a/gemrb/core/Interface.cpp b/gemrb/core/Interface.cpp
index b09dccdc4..4a0edad26 100644
--- a/gemrb/core/Interface.cpp
+++ b/gemrb/core/Interface.cpp
@@ -784,13 +784,13 @@ void Interface::HandleFlags() noexcept
 	if (QuitFlag&QF_LOADGAME) {
 		QuitFlag &= ~QF_LOADGAME;
 
+		winmgr->CloseAllWindows();
 		LoadGame(LoadGameIndex, VersionOverride );
 		LoadGameIndex.reset();
 		//after loading a game, always check if the game needs to be upgraded
 	}
 
 	if (QuitFlag&QF_ENTERGAME) {
-		winmgr->CloseAllWindows();
 		QuitFlag &= ~QF_ENTERGAME;
 		if (game) {
 			EventFlag|=EF_EXPANSION;
```

> Ideally game clock should start to tick at the exact same moment.

this sounds completely unrelated

**Comment by burner1024** [comment]
> does this only happen when loading a game from within another game?

Here's what happens when loading the game from fresh start.
![Captura desde 2024-02-14 13-17-00](https://github.com/gemrb/gemrb/assets/1279852/809d0279-3375-4b31-88fb-5c25437ccc2a)

> this sounds completely unrelated

Why?

**Comment by bradallred** [comment]
That should be easily fixed by just relocating `GemRB.EnterGame` to `EndLoadScreen` right before we close it.

> Why?

because time keeping and GUI presentation are unrelated.

**Comment by burner1024** [comment]
That seems very strange. I think they should be related.
From player's point of view, game going on without displaying it is just chunks of time going missing.

**Comment by bradallred** [comment]
right, but displaying the GUI isn't what advances time; its just a display. In fact, I _thought_ that opening the inventory was supposed to pause the game (eg stop time). But I suppose with this issue closing it would unpause it. Even then, the issue would be that the pause/unpause functions are allowed manipulate the game clock while the game is loading which, as I said, is a separate issue.

**Comment by bradallred** [comment]
I'll have a fix for this in #1876

**Comment by burner1024** [comment]
I don't entirely agree, but yes, let's see how it works after that PR is merged.
